epam_tr/src/main/java/com/epam/tr/service/FileSystemService.java

```diff
@@ -0,0 +1,141 @@
+package com.epam.tr.service;
+
+import com.epam.tr.dto.FileDto;
+import com.epam.tr.entities.FileSystemObjectType;
+import com.epam.tr.service.logic.builder.FileEntityBuilder;
+import com.epam.tr.service.logic.sorter.SorterFactory;
+import com.epam.tr.service.logic.validator.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.epam.tr.entities.FileSystemObjectType.*;
+import static java.io.File.listRoots;
+import static java.util.Arrays.stream;
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class FileSystemService implements FileService {
+
+    @Autowired
+    private Validator<FileDto> validator;
+    private FileEntityBuilder builder = new FileEntityBuilder();
+
+    @Override
+    public void create(String drive, MultiValueMap<String, String> allRequestParams) throws IOException {
+        FileDto fileDto = builder.buildFileEntity(drive, allRequestParams);
+        if (validator.isValid(fileDto)) {
+            if (fileDto.getType() == FOLDER) {
+                createFolder(fileDto);
+            }
+            if (fileDto.getType() == FILE) {
+                createFile(fileDto);
+            }
+        }
+    }
+
+    @Override
+    public void update(FileDto oldFile, FileDto newFile) throws IOException {
+        if (validator.isValid(newFile)) {
```

Comment:
sometimes it is ok.
but nesting is not a good thing, it makes code hard to read.
you can avoid nesting here like: 
`if (!validator.isValid(newFile))  return`